message/message-factory.rst

```diff
@@ -1,57 +1,124 @@
 .. _message-factory:
+.. _stream-factory:
 
-Message Factory
-===============
+HTTP Factories
```

Comment:
i adjusted some of the naming. in many places, we do explicitly mean a message factory and not any http factory.